### PR TITLE
fix(chunks) - backwards compatible automatic forwarding of parts to shard trackers

### PR DESCRIPTION
**Pull request by birchmd**
Brings the logic originally merged in https://github.com/near/nearcore/pull/2822 into master in a backwards compatible way.

This logic makes nodes automatically forward chunk parts to the owners, thus removing an unnecessary request/response cycle (see https://github.com/near/nearcore/issues/2608 for more information).

Fixes: https://github.com/near/nearcore/issues/3347

Testing plan: Tests from the original PR.

### CODE REVIEW

File: chain/chunks/src/lib.rs [hunk 1]
```diff
@@ -979,6 +1037,39 @@ impl ShardsManager {
...
+            }
+
+            Some(existing_parts) => {
+                for part in forward.parts {
+                    let part_ord = part.part_ord;
+                    if part_ord > num_total_parts {
+                        warn!(target: "chunks", "Received chunk part with part_ord greater than the the total number of chunks");
+                        continue;
+                    }
+                    if let Some(_) = existing_parts.insert(part_ord, part) {
```
Comment (bowenwang1996) [inline chain/chunks/src/lib.rs:1065]:
```suggestion
                    if existing_parts.insert(part_ord, part).is_some() {
```

File: chain/client/src/client.rs [hunk 2]
```diff
@@ -40,6 +41,7 @@ use crate::metrics;
 use crate::sync::{BlockSync, HeaderSync, StateSync, StateSyncResult};
 use crate::types::{Error, ShardSyncDownload};
 use crate::SyncStatus;
+use near_primitives::version::{ProtocolVersion, PROTOCOL_VERSION};
```
Comment (frol) [inline chain/client/src/client.rs:44]: nit: move the import to the group above.

**Review by frol — APPROVED** [review]
The changes in `neard` and `primitives` LGTM

**Comment by birchmd** [comment]
Closing this PR in favour of one that uses a branch from the main repo.

**Pull request closed without merging**